chain/chain/src/runtime/mod.rs

```diff
@@ -806,6 +818,23 @@ impl RuntimeAdapter for NightshadeRuntime {
             if let Some(iter) = transaction_groups.next() {
                 while let Some(tx) = iter.next() {
                     num_checked_transactions += 1;
+
+                    if ProtocolFeature::CongestionControl.protocol_version() <= protocol_version {
```

Comment:
I was told to avoid compile time feature flags, so when I removed that from the initial implementation, I also removed calls to `checked_feature!`. I don't see the point of the macro without compile time flags. It just raises the bar in Rust proficiency for anyone to understand what the code does.

How about this? I can add a method `ProtocolFeature::enabled(protocol_version)` that makes it less error prone, and then replace the comparisons with calls to that? We could even go as far and replace all invocations of `checked_feature("stable", $($arg:tt)*)` if we want to.

In any case, I think this warrants a separate PR.

Reply:
Here is the follow-up PR for this check: https://github.com/near/nearcore/pull/11237